src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public class FigureSupplier extends Circle {
```

Comment:
This is completely wrong.
1. Why does the supplier extends the figure. Either you have a figure or a figure supplier.
2. Why does your supplier "supplies" only the specific figure - this makes it a circle supplier, not a figure supplier.